### PR TITLE
Cox Regression/Survival Forest: Date-to-rate and rate-to-date survival prediction

### PR DESCRIPTION
**Pull request by hideaki**
# Description
Date-to-rate and rate-to-date survival prediction.

# Checklist
Make sure you have performed the following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/survival_forest.R [hunk 1]
```diff
@@ -603,8 +612,52 @@ glance.ranger_survival_exploratory <- function(x, data_type = "training", ...) {
   }
 }
 
+survival_time_to_predicted_rate <- function(survival_mat, time_vec, time_index_fun) {
```
Comment (hidekoji) [inline R/survival_forest.R:615]: Add comment for arguments.

File: R/survival_forest.R [hunk 2]
```diff
@@ -619,6 +672,13 @@ augment.ranger_survival_exploratory <- function(x, newdata = NULL, data_type = "
     }
 
     predictor_variables <- attr(x$df,"predictors")
+    # If start/end time column is in newdata, use it.
+    if (!is.null(x$clean_start_time_col) && x$terms_mapping[x$clean_start_time_col] %in% colnames(newdata)) {
+      predictor_variables <- c(predictor_variables, x$clean_start_time_col)
+    }
+    if (!is.null(x$clean_end_time_col) && x$terms_mapping[x$clean_end_time_col] %in% colnames(newdata)) {
```
Comment (hidekoji) [inline R/survival_forest.R:679]: This might be not needed.

File: R/build_coxph.R [hunk 3]
```diff
@@ -87,18 +96,48 @@ partial_dependence.coxph_exploratory <- function(fit, time_col, vars = colnames(
     # 3    12    224       1        0      0.969      0.972      0.977      0.977      0.975      0.969
     # 4    13    223       2        0      0.957      0.961      0.968      0.967      0.966      0.957
 
-    # TODO: There is info on confidence interval, but we are not making use of them here.
-    est <- as.data.frame(t(res %>% select(starts_with('estimate.'))))
+    # Adjuct values in time with approxfun. WIP.
```
Comment (hidekoji) [inline R/build_coxph.R:99]: Remove wip

File: R/survival_forest.R [hunk 4]
```diff
@@ -612,14 +612,23 @@ glance.ranger_survival_exploratory <- function(x, data_type = "training", ...) {
   }
 }
 
+# Extracts and interpolates survival rate from survival_mat, for the time specified for each observation by time_vec. 
+# survival_mat - Predicted survival curve data obtained from the prediction. 
+# time_vec - Time to extract the prediction for from the survival_mat for each observation.
```
Comment (hidekoji) [inline R/survival_forest.R:617]: Grammarly complains about the `for from` so this could be `for`, I think.

File: R/survival_forest.R [hunk 5]
```diff
@@ -603,8 +612,61 @@ glance.ranger_survival_exploratory <- function(x, data_type = "training", ...) {
...
```
Comment (hideaki) [inline R/survival_forest.R:617]:
```suggestion
# time_vec - Time to extract the prediction from the survival_mat for each observation.
```

**Pull request merged**